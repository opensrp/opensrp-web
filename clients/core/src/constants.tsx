// endpoints
export const OPENSRP_CLIENT_ENDPOINT = 'client/searchByCriteria';
export type OPENSRP_CLIENT_ENDPOINT = typeof OPENSRP_CLIENT_ENDPOINT;
export const OPENSRP_MANIFEST_ENDPOINT = 'manifest';
export type OPENSRP_MANIFEST_ENDPOINT = typeof OPENSRP_MANIFEST_ENDPOINT;
export const OPENSRP_FORMS_ENDPOINT = 'clientForm';
export type OPENSRP_FORMS_ENDPOINT = typeof OPENSRP_FORMS_ENDPOINT;
export const OPENSRP_MANIFEST_FORMS_ENDPOINT = 'clientForm/release-related-files';
export type OPENSRP_MANIFEST_FORMS_ENDPOINT = typeof OPENSRP_MANIFEST_FORMS_ENDPOINT;
export const OPENSRP_FORM_METADATA_ENDPOINT = 'clientForm/metadata';
export type OPENSRP_FORM_METADATA_ENDPOINT = typeof OPENSRP_FORM_METADATA_ENDPOINT;

// internal urls
export const LOGIN_URL = '/login';
export type LOGIN_URL = typeof LOGIN_URL;
export const LOGOUT_URL = '/logout';
export type LOGOUT_URL = typeof LOGOUT_URL;
export const HOME_URL = '/';
export type HOME_URL = typeof HOME_URL;
export const CLIENT_URL = '/clients';
export type CLIENT_URL = typeof CLIENT_URL;
export const CLIENT_PROFILE_URL = '/client-profile';
export type CLIENT_PROFILE_URL = typeof CLIENT_PROFILE_URL;
export const HOUSEHOLD_URL = '/404';
export type HOUSEHOLD_URL = typeof HOUSEHOLD_URL;
export const ANC_URL = '/404';
export type ANC_URL = typeof ANC_URL;
export const CHILD_URL = '/child';
export type CHILD_URL = typeof CHILD_URL;
export const CHILD_PROFILE_URL = '/child-profile';
export type CHILD_PROFILE_URL = typeof CHILD_PROFILE_URL;
export const USER_URL = '/404';
export type USER_URL = typeof USER_URL;
export const ROLE_URL = '/404';
export type ROLE_URL = typeof ROLE_URL;
export const TEAM_URL = '/404';
export type TEAM_URL = typeof TEAM_URL;
export const LOCATIONS_URL = '/404';
export type LOCATION_URL = typeof LOCATIONS_URL;
export const REPORTS_URL = '/404';
export type REPORTS_URL = typeof REPORTS_URL;
export const VIEW_RELEASE_PAGE_URL = '/manifest/releases';
export type VIEW_RELEASE_PAGE_URL = typeof VIEW_RELEASE_PAGE_URL;
export const VIEW_DRAFT_FILES_PAGE_URL = '/files/draft';
export type VIEW_DRAFT_FILES_PAGE_URL = typeof VIEW_DRAFT_FILES_PAGE_URL;
export const VIEW_JSON_VALIDATORS_PAGE_URL = '/json-validators';
export type VIEW_JSON_VALIDATORS_PAGE_URL = typeof VIEW_JSON_VALIDATORS_PAGE_URL;
export const MANIFEST_FILE_UPLOAD = '/manifest';
export type MANIFEST_FILE_UPLOAD = typeof MANIFEST_FILE_UPLOAD;

// string literals

export const CLIENT = 'Client';
export type CLIENT = typeof CLIENT;
export const REPORT = 'Report';
export type REPORT = typeof REPORT;
export const ADMIN = 'Admin';
export type ADMIN = typeof ADMIN;
export const CLIENT_RECORDS = 'Client Records';
export type CLIENT_RECORDS = typeof CLIENT_RECORDS;
export const REPORTS = 'Reports';
export type REPORTS = typeof REPORTS;
export const ALL_CLIENTS = 'All clients';
export type ALL_CLIENTS = typeof ALL_CLIENTS;
export const HOUSEHOLD = 'Household';
export type HOUSEHOLD = typeof HOUSEHOLD;
export const ANC = 'ANC';
export type ANC = typeof ANC;
export const CHILD = 'Child';
export type CHILD = typeof CHILD;
export const USERS = 'Users';
export type USERS = typeof USERS;
export const ROLES = 'Roles';
export type ROLES = typeof ROLES;
export const TEAMS = 'Teams';
export type TEAMS = typeof TEAMS;
export const LOCATIONS = 'Locations';
export type LOCATIONS = typeof LOCATIONS;
export const LOGIN = 'Login';
export type LOGIN = typeof LOGIN;
export const SIGN_OUT = 'Sign Out';
export type SIGN_OUT = typeof SIGN_OUT;
export const CHILD_CLIENT_TYPE = 'child';
export type CHILD_CLIENT_TYPE = typeof CHILD_CLIENT_TYPE;
export const VIEW_RELEASE = 'View Release';
export type VIEW_RELEASE = typeof VIEW_RELEASE;
export const VIEW_JSON_VALIDATORS = 'View JSON Validators';
export type VIEW_JSON_VALIDATORS = typeof CHILD_CLIENT_TYPE;
export const VIEW_DRAFT_FILES = 'View Draft Files';
export type VIEW_DRAFT_FILES = typeof VIEW_DRAFT_FILES;
export const FILE_UPLOAD_TYPE = 'file-upload';
export type FILE_UPLOAD_TYPE = typeof FILE_UPLOAD_TYPE;
export const VALIDATOR_UPLOAD_TYPE = 'validator_upload';
export type VALIDATOR_UPLOAD_TYPE = typeof VALIDATOR_UPLOAD_TYPE;
export const JSON_VALIDATOR_LABLE = 'JSON Validators';
export type JSON_VALIDATOR_LABLE = typeof JSON_VALIDATOR_LABLE;
export const DRAFT_FILE_LABEL = 'Draft Files';
export type DRAFT_FILE_LABEL = typeof DRAFT_FILE_LABEL;
export const RELEASE_FILE_LABEL = 'Release';
export type RELEASE_FILE_LABEL = typeof RELEASE_FILE_LABEL;
export const RELEASES_LABEL = 'Releases';
export type RELEASES_LABEL = typeof RELEASES_LABEL;
export const EDIT_FILE_LABEL = 'Edit Form';
export type EDIT_FILE_LABEL = typeof EDIT_FILE_LABEL;
export const UPLOAD_FILE_LABEL = 'Upload Form';
export type UPLOAD_FILE_LABEL = typeof UPLOAD_FILE_LABEL;
export const DOWNLOAD_LABEL = 'Download';
export type DOWNLOAD_LABEL = typeof DOWNLOAD_LABEL;
export const MAKE_RELEASE_LABEL = 'Make Release';
export type MAKE_RELEASE_LABEL = typeof MAKE_RELEASE_LABEL;
export const FILE_NAME_LABEL = 'File Name';
export type FILE_NAME_LABEL = typeof FILE_NAME_LABEL;
export const FILE_VERSION_LABEL = 'File Version';
export type FILE_VERSION_LABEL = typeof FILE_VERSION_LABEL;
export const IDENTIFIER_LABEL = 'Identifier';
export type IDENTIFIER_LABEL = typeof IDENTIFIER_LABEL;
export const MODULE_LABEL = 'Module';
export type MODULE_LABEL = typeof MODULE_LABEL;
export const EDIT_LABEL = 'Edit';
export type EDIT_LABEL = typeof EDIT_LABEL;
export const UPLOAD_EDIT_LABEL = 'Upload edit';
export type UPLOAD_EDIT_LABEL = typeof UPLOAD_EDIT_LABEL;
export const UPOL0AD_FILE_LABEL = 'Upload New File';
export type UPOL0AD_FILE_LABEL = typeof UPOL0AD_FILE_LABEL;
export const VIEW_FILES_LABEL = 'View Files';
export type VIEW_FILES_LABEL = typeof VIEW_FILES_LABEL;
export const APP_ID_LABEL = 'APP Id';
export type APP_ID_LABEL = typeof APP_ID_LABEL;
export const APP_VERSION_LABEL = 'App Version';
export type APP_VERSION_LABEL = typeof APP_VERSION_LABEL;
export const RELATED_TO_LABEL = 'Related to';
export type RELATED_TO_LABEL = typeof RELATED_TO_LABEL;
export const FILE_UPLOAD_LABEL = 'Upload file';
export type FILE_UPLOAD_LABEL = typeof FILE_UPLOAD_LABEL;
export const FORM_REQUIRED_LABEL = 'Form is required';
export type FORM_REQUIRED_LABEL = typeof FORM_REQUIRED_LABEL;
export const FORM_NAME_REQUIRED_LABEL = 'Form name is required';
export type FORM_NAME_REQUIRED_LABEL = typeof FORM_NAME_REQUIRED_LABEL;
export const FIND_DRAFT_RELEASES_LABEL = 'Find Draft Files';
export type FIND_DRAFT_RELEASES_LABEL = typeof FIND_DRAFT_RELEASES_LABEL;
export const FIND_FILES_LABEL = 'Find Files';
export type FIND_FILES_LABEL = typeof FIND_FILES_LABEL;
export const FIND_RELEASES_LABEL = 'Find Release';
export type FIND_RELEASES_LABEL = typeof FIND_RELEASES_LABEL;
export const SEARCH_LABEL = 'Search';
export type SEARCH_LABEL = typeof SEARCH_LABEL;
export const CREATED_AT_LABEL = 'Created at';
export type CREATED_AT_LABEL = typeof CREATED_AT_LABEL;
export const UPDATED_AT_LABEL = 'Updated at';
export type UPDATED_AT_LABEL = typeof UPDATED_AT_LABEL;
export const FIND_DRAFT_FILE_LABEL = 'Find Draft Files';
export type FIND_DRAFT_FILE_LABEL = typeof FIND_DRAFT_FILE_LABEL;
export const FIND_RELEASE_FILES = 'Find Release Files';
export type FIND_RELEASE_FILES = typeof FIND_RELEASE_FILES;
export const FIND_VALIDATOR_FILES = 'Find Validator Files';
export type FIND_VALIDATOR_FILES = typeof FIND_VALIDATOR_FILES;

// url constants
export const PAGINATION_SIZE = 10;
export type PAGINATION_SIZE = typeof PAGINATION_SIZE;
export const PAGINATION_NEIGHBORS = 3;
export type PAGINATION_NEIGHBORS = typeof PAGINATION_NEIGHBORS;
export const NEXT = 'Next';
export type NEXT = typeof NEXT;
export const START = 'Start';
export type START = typeof START;
export const LAST = 'Last';
export type LAST = typeof LAST;
export const PREVIOUS = 'Previous';
export type PREVIOUS = typeof PREVIOUS;
export const LOADING = 'Loading';
export const VIEW = 'View';
export type VIEW = typeof VIEW;
