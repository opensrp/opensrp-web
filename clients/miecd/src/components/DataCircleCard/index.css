.compartment-wrapper .circlesRow .circle {
  list-style-type: none;
  width: 7.5rem;
  height: 7.5rem;
  border-radius: 3.75rem;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.compartment-wrapper .circlesRow .circle:hover {
  cursor: pointer;
}

.compartment-wrapper .card-body > p {
  margin-top: 12%;
}

.compartment-wrapper .dataCircleCard {
  padding: 0;
  background-color: #ffffff;
  border: 0.063em solid #dadce0;
  border-radius: 0.313rem;
  box-shadow: 0 0.063em 0.188em 0 #d7d7e0;
  height: 14.375rem;
}

.compartment-wrapper .circlesRow {
  display: flex;
  justify-content: space-around;
  align-items: center;
}

.compartment-wrapper .card-group > .card {
  flex: 1 1 auto;
}

.compartment-wrapper .card {
  border: 0px;
  margin: 20px 50px 20px 0;
}

.compartment-wrapper .circlesRow > .red {
  background-color: #ff1e1e;
}

.compartment-wrapper .circlesRow>.yellow {
    background-color: #FFD300;
}

.compartment-wrapper .circlesRow > .green {
  background-color: #37ae12;
}

.compartment-wrapper .circlesRow > .total-children {
  background-color: #637381;
}

.compartment-wrapper .circlesRow > .stunting {
  background-color: #ff6361;
}

.compartment-wrapper .circlesRow > .wasting {
  background-color: #003f5c;
}

.compartment-wrapper .circlesRow > .overweight {
  background-color: #58508d;
}

.compartment-wrapper .circlesRow > .inappropriate-feeding {
  background-color: #003f5c;
}

.compartment-wrapper .circlesRow > .green,
.compartment-wrapper .circlesRow > .red,
.compartment-wrapper .circlesRow > .orange {
  border: none !important;
}

.compartments .card-title {
  border-bottom: 0.063em solid #dadce0;
  color: #001429;
  font-size: 1.125rem;
  font-weight: 400;
  line-height: 1.375rem;
  text-align: left;
  padding: 1rem;
}

.compartment-wrapper span.number {
  color: #ffffff;
  font-size: 1.25rem;
  font-weight: 400;
  line-height: 0.875rem;
}

.compartment-wrapper span.risk-level {
  color: #ffffff;
  font-size: 0.75rem;
  font-weight: 400;
  line-height: 0.875rem;
}

.compartment-wrapper .circleRow > a,
.compartment-wrapper a:hover {
  color: inherit;
  text-decoration: inherit;
}

@media (max-width: 1365px) {
}
