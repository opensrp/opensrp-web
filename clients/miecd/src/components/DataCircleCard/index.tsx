import React from 'react';
import { connect } from 'react-redux';
import { Link, RouteComponentProps, withRouter } from 'react-router-dom';
import { Card, CardBody, CardTitle } from 'reactstrap';
import {
    ALL,
    INAPPROPRIATELY_FED,
    NBC_AND_PNC_CHILD,
    NBC_AND_PNC_WOMAN,
    NO,
    NO_RISK,
    NUTRITION,
    OVERWEIGHT,
    PREGNANCY,
    RED,
    RED_ALERT,
    RISK,
    SEVERE_WASTING,
    SMS_FILTER_FUNCTION,
    STUNTED,
} from '../../constants';
import { getLinkToHierarchicalDataTable } from '../../helpers/utils';
import { FlexObject } from '../../helpers/utils';
import { addFilterArgs } from '../../store/ducks/sms_events';
import './index.css';

/**
 * interface for props to be passed to DataCircleCard component.
 */
interface Props extends RouteComponentProps {
    redAlert?: number;
    risk?: number;
    noRisk?: number;
    totalChildren?: number;
    stunting?: number;
    wasting?: number;
    overweight?: number;
    inappropriateFeeding?: number;
    title: string;
    addFilterArgsActionCreator?: typeof addFilterArgs;
    filterArgs?: SMS_FILTER_FUNCTION[];
    module: PREGNANCY | NBC_AND_PNC_CHILD | NBC_AND_PNC_WOMAN | NUTRITION | '';
    className?: string;
    userLocationId: string;
    permissionLevel: number;
}

/**
 * functional component that takes in props
 */
interface CircleSpecProps {
    class: string;
    riskLabel: string;
    riskType: string;
    riskValue: any;
}

function DataCircleCard({
    redAlert,
    risk,
    noRisk,
    stunting,
    wasting,
    overweight,
    inappropriateFeeding,
    title,
    addFilterArgsActionCreator = addFilterArgs,
    filterArgs,
    module,
    className = '',
    userLocationId,
    permissionLevel,
    history,
}: Props) {
    const pregnancyAndPncCircleSpec: CircleSpecProps[] = [
        {
            class: 'red',
            riskLabel: RED_ALERT,
            riskType: RED,
            riskValue: redAlert,
        },
        {
            class: 'yellow',
            riskLabel: RISK,
            riskType: RISK,
            riskValue: risk,
        },
        {
            class: 'green',
            riskLabel: NO_RISK,
            riskType: NO,
            riskValue: noRisk,
        },
    ];

    const nutritionCircleSpec: CircleSpecProps[] = [
        {
            class: 'stunting',
            riskLabel: 'Stunting',
            riskType: STUNTED,
            riskValue: stunting,
        },
        {
            class: 'wasting',
            riskLabel: 'Wasting',
            riskType: SEVERE_WASTING,
            riskValue: wasting,
        },
        {
            class: 'overweight',
            riskLabel: 'Overweight',
            riskType: OVERWEIGHT,
            riskValue: overweight,
        },
        {
            class: 'inappropriate-feeding',
            riskLabel: 'Inappropriate Feeding',
            riskType: INAPPROPRIATELY_FED,
            riskValue: inappropriateFeeding,
        },
    ];

    return (
        <Card className={`dataCircleCard ${className}`}>
            <CardTitle>
                <Link
                    to={getLinkToHierarchicalDataTable(ALL, module, title, permissionLevel, userLocationId)}
                    // tslint:disable-next-line: jsx-no-lambda
                    onClick={() => {
                        if (filterArgs) {
                            addFilterArgsActionCreator(filterArgs);
                        }
                    }}
                >
                    {title}
                </Link>
            </CardTitle>
            <CardBody className="circlesRow">
                {(module !== NUTRITION ? pregnancyAndPncCircleSpec : nutritionCircleSpec).map(
                    (spec: FlexObject, i: number) => (
                        <Link
                            className={`${spec.class} circle`}
                            key={i}
                            // tslint:disable-next-line: jsx-no-lambda
                            onClick={() => {
                                if (filterArgs) {
                                    addFilterArgsActionCreator(filterArgs);
                                }
                            }}
                            to={getLinkToHierarchicalDataTable(
                                spec.riskType,
                                module,
                                title,
                                permissionLevel,
                                userLocationId,
                            )}
                        >
                            <span className="number">{spec.riskValue}</span>
                            <span className="risk-level">{spec.riskLabel}</span>
                        </Link>
                    ),
                )}
            </CardBody>
        </Card>
    );
}

const mapDispatchToProps = { addFilterArgsActionCreator: addFilterArgs };

const ConnectedDataCircleCard = connect(null, mapDispatchToProps)(withRouter(DataCircleCard));

export default ConnectedDataCircleCard;
