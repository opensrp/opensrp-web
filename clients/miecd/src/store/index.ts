import { connectReducer, getConnectedStore } from '@onaio/connected-reducer-registry';
import { gateKeeperReducer, gateKeeperReducerName } from '@onaio/gatekeeper';
import reducerRegistry, { combine, Registry } from '@onaio/redux-reducer-registry';
import session, { reducerName as sessionReducer } from '@onaio/session-reducer';

/** Initial reducers in the reducer registry */
const defaultReducers: Registry = {
    router: connectReducer as any /** Dirty hack because connectRouter LocationChangeAction does not extend Redux.AnyAction */,
};

/** Add users reducer to registry */
defaultReducers[sessionReducer] = session;
defaultReducers[gateKeeperReducerName] = gateKeeperReducer;

/** The initial store for the reveal web app */
const store = getConnectedStore(defaultReducers);

/** Set listener to add reducers to store when registered */
reducerRegistry.setChangeListener(reducers => {
    store.replaceReducer(combine(reducers));
});

export default store;
