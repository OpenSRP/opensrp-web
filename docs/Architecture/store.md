# overview

This documentation will take you through the process of creating and working with:

    - reducers
    - actions
    - selectors

## Where do I find the store

the store reducer modules are found within src/store. The initial directory structure looks something like this: (Note: this structure only depicts the initial directory structure and will as expected contain more directories and files as more are added with time)

```
...
├── src
│   ├── ...
│   ├── store
│   │   ├── ducks
│   │   │    ├── superset.ts
│   │   │    ├── tests/
│   │   ├── tests
│   │   ├── index.ts
│   │   └── selectors.ts
...
```

A single reducer module will contain the afore mentioned parts i.e. its respective actions, action creators if any, its selectors and the reducer function

## Actions

An action can be any object with a type property, they are the only way provisioned for changing the state in a redux store.

### Adding Actions

Its a common convention as well as recommended one to Use type constants, [read more on this here]().

steps:

- Add An interface describing the action structure: redux exposes an `AnyAction` type that can be extended to write custom action interfaces
- Create a general actiontypes to be used to type check all actions that will be passed to the reducer function as an argument. this will be in the form of [intersection type]()

## Selectors

whereas the action puts data in the store; the selectors on the other hand act as getters of the slice of data that this reducer module handles

selectors are functions that will take at least one argument, the store, Then based on any other arguments like say an id, can be used to filter down the data retrieved.

## The Reducer

The reducer function, takes 2 arguments; the state and an action.
using the type property of the action, the reducer should then know how to modify the state.
such functionality could be implemented as a form of switch strcture such as a list of if statements or the more recommended approach which is to use the actual switch statement

### Adding the Reducer

- Define an interface for the data to be stored in the store
- Define an interface for the store state.
- write the actual reducer function, while applying the required type annotations as required.
