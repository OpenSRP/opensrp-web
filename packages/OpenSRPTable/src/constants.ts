// string literals
export const PLEASE_DEFINE_SOME_COLUMNS = 'Please define some columns';
export type PLEASE_DEFINE_SOME_COLUMNS = typeof PLEASE_DEFINE_SOME_COLUMNS;
