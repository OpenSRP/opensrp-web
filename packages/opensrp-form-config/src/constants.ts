export const MAKE_RELEASE_LABEL = 'Make Release';
export const FILE_NAME_LABEL = 'File Name';
export const FILE_VERSION_LABEL = 'File Version';
export const IDENTIFIER_LABEL = 'Identifier';
export const MODULE_LABEL = 'Module';
export const EDIT_LABEL = 'Edit';
export const DOWNLOAD_LABEL = 'Download';
export const UPLOAD_EDIT_LABEL = 'Upload edit';
