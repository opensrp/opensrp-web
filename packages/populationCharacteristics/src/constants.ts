export const SEARCH_LABEL = 'Search';
export const SEARCH_SETTINGS_LABEL = 'Search Settings';
export const PAGE_TITLE_LABEL = 'Server Settings';
export const EDIT_LABEL = 'Edit';
export const DESCRIPTION_LABEL = 'Description';
export const SETTINGS_LABEL = 'Settings';
export const INHERITED_FROM_LABEL = 'Inherited from';
export const NAME_LABEL = 'Name';
export const SET_TO_YES_LABEL = `Set to 'Yes'`;
export const SET_TO_NO_LABEL = `Set to 'No'`;
export const INHERIT_SETTING_LABEL = 'Inherit setting';
export const NO_DATA_FOUND = 'No data found';
