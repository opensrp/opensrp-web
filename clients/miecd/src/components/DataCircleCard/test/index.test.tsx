import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { createBrowserHistory } from 'history';
import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import DataCircleCard from '..';
import { PREGNANCY } from '../../../constants';
import store from '../../../store/index';

const history = createBrowserHistory();

describe('DataCircleCard', () => {
    it('must render without crashing', () => {
        const props = { highRisk: 10, lowRisk: 10, noRisk: 10, title: 'test title' };
        shallow(
            <Provider store={store}>
                <Router history={history}>
                    <DataCircleCard {...props} module={PREGNANCY} />
                </Router>
            </Provider>,
        );
    });
    it('must render correctly', () => {
        const props = { highRisk: 10, lowRisk: 10, noRisk: 10, title: 'test title' };
        const wrapper = mount(
            <Provider store={store}>
                <Router history={history}>
                    <DataCircleCard {...props} module={PREGNANCY} />
                </Router>
            </Provider>,
        );
        expect(toJson(wrapper.find('Card'))).toMatchSnapshot();
    });
});
