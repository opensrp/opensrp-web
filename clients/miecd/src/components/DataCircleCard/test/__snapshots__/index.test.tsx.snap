// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataCircleCard must render correctly 1`] = `
<Card
  className="dataCircleCard "
  tag="div"
>
  <div
    className="dataCircleCard  card"
  >
    <CardTitle
      tag="div"
    >
      <div
        className="card-title"
      >
        <Link
          onClick={[Function]}
          to="/pregnancy_compartments/hierarchicaldata/Pregnancy/all/test title/undefined/down/undefined/undefined"
        >
          <LinkAnchor
            href="/pregnancy_compartments/hierarchicaldata/Pregnancy/all/test title/undefined/down/undefined/undefined"
            navigate={[Function]}
            onClick={[Function]}
          >
            <a
              href="/pregnancy_compartments/hierarchicaldata/Pregnancy/all/test title/undefined/down/undefined/undefined"
              onClick={[Function]}
            >
              test title
            </a>
          </LinkAnchor>
        </Link>
      </div>
    </CardTitle>
    <CardBody
      className="circlesRow"
      tag="div"
    >
      <div
        className="circlesRow card-body"
      >
        <Link
          className="red circle"
          key="0"
          onClick={[Function]}
          to="/pregnancy_compartments/hierarchicaldata/Pregnancy/red/test title/undefined/down/undefined/undefined"
        >
          <LinkAnchor
            className="red circle"
            href="/pregnancy_compartments/hierarchicaldata/Pregnancy/red/test title/undefined/down/undefined/undefined"
            navigate={[Function]}
            onClick={[Function]}
          >
            <a
              className="red circle"
              href="/pregnancy_compartments/hierarchicaldata/Pregnancy/red/test title/undefined/down/undefined/undefined"
              onClick={[Function]}
            >
              <span
                className="number"
              />
              <span
                className="risk-level"
              >
                Red Alert
              </span>
            </a>
          </LinkAnchor>
        </Link>
        <Link
          className="yellow circle"
          key="1"
          onClick={[Function]}
          to="/pregnancy_compartments/hierarchicaldata/Pregnancy/risk/test title/undefined/down/undefined/undefined"
        >
          <LinkAnchor
            className="yellow circle"
            href="/pregnancy_compartments/hierarchicaldata/Pregnancy/risk/test title/undefined/down/undefined/undefined"
            navigate={[Function]}
            onClick={[Function]}
          >
            <a
              className="yellow circle"
              href="/pregnancy_compartments/hierarchicaldata/Pregnancy/risk/test title/undefined/down/undefined/undefined"
              onClick={[Function]}
            >
              <span
                className="number"
              />
              <span
                className="risk-level"
              >
                risk
              </span>
            </a>
          </LinkAnchor>
        </Link>
        <Link
          className="green circle"
          key="2"
          onClick={[Function]}
          to="/pregnancy_compartments/hierarchicaldata/Pregnancy/no/test title/undefined/down/undefined/undefined"
        >
          <LinkAnchor
            className="green circle"
            href="/pregnancy_compartments/hierarchicaldata/Pregnancy/no/test title/undefined/down/undefined/undefined"
            navigate={[Function]}
            onClick={[Function]}
          >
            <a
              className="green circle"
              href="/pregnancy_compartments/hierarchicaldata/Pregnancy/no/test title/undefined/down/undefined/undefined"
              onClick={[Function]}
            >
              <span
                className="number"
              >
                10
              </span>
              <span
                className="risk-level"
              >
                No Risk
              </span>
            </a>
          </LinkAnchor>
        </Link>
      </div>
    </CardBody>
  </div>
</Card>
`;
